### ISSUE TITLE
NCCL 2.9.6 builds fail with some undefined references

### ISSUE THREAD

**Issue by ShadowTemplate** [NVIDIA/nccl#510]
Hi everyone!
Today I have received an error while building [XGBoost](https://github.com/dmlc/xgboost) (v1.4.1) in this environment:
- CUDA 11.2.152
- NCCL 2.9.6
- GPU_COMPUTE_VER 52;60;61;62;70;72;75;80;86
- CMake 3.18.4
- GCC 8.4.0

```
...
[ 99%] Linking CXX executable ../xgboost
/cm/shared/apps/nccl2-cuda11.2-gcc8/current/lib/libnccl_static.a(enqueue.o): In function `ncclGetCudaGraph(ncclComm*, CUgraph_st**)':
(.text+0x30eb): undefined reference to `cudaStreamGetCaptureInfo_v2'
/cm/shared/apps/nccl2-cuda11.2-gcc8/current/lib/libnccl_static.a(enqueue.o): In function `ncclCudaGraphHostSetup(ncclComm*, CUgraph_st*)':
(.text+0x31ee): undefined reference to `cudaUserObjectCreate'
(.text+0x320c): undefined reference to `cudaGraphRetainUserObject'
(.text+0x3268): undefined reference to `cudaStreamUpdateCaptureDependencies'
/cm/shared/apps/nccl2-cuda11.2-gcc8/current/lib/libnccl_static.a(enqueue.o): In function `ncclEnqueueCheck(ncclInfo*)':
(.text+0x3655): undefined reference to `cudaStreamGetCaptureInfo_v2'
collect2: error: ld returned 1 exit status
make[2]: *** [CMakeFiles/runxgboost.dir/build.make:291: ../xgboost] Error 1
make[1]: *** [CMakeFiles/Makefile2:203: CMakeFiles/runxgboost.dir/all] Error 2
make: *** [Makefile:149: all] Error 2
```

I suspect the issue is not related to XGBoost, since I have no issue with NCCL 2.8.4 (and the same environment).
Is it possible that some references are indeed missing in the NCCL library provided by NVIDIA?

Thanks a lot.

Kind regards,
GT

**Comment by sjeaugey** [comment]
It looks like you are using a NCCL version built for CUDA 11.3, hence the errors, since `cudaStreamGetCaptureInfo_v2` is not available on CUDA 11.2.

Can you upgrade to CUDA 11.3, or use a NCCL version compiled against CUDA 11.0 and see if it fixes the issue?

**Comment by ShadowTemplate** [comment]
Thanks a lot for your help.
You are right and I was indeed using a CUDA 11.3-based version of NCCL.

As I wrote earlier, I have no issue with NCCL 2.8.4 for CUDA 11.2. I will use this one.

Sorry for the mess!
Have a great day.
